help: Document "image viewer" mobile feature.

drrosa opened the pull request:
- Updates [View and browse images](https://zulip.com/help/view-and-browse-images), adding desktop/web app instructions and tweaking the intro paragraph.
- Renames page to **View images and videos**.
- Clarifies viewer support for browsing both images and videos, and tweaks documentation on other pages.
- Adds a new section to describing the video player features.
- Splits out browsing other images and videos in the view into its own section.
- Adds mobile app instructions.

Fixes #27801.

**Screenshots and screen captures:**
- https://zulip.com/help/view-and-browse-images
![image](https://github.com/zulip/zulip/assets/2343554/25cb8fda-9e07-42ad-909d-add8d42c5751)
![image](https://github.com/zulip/zulip/assets/2343554/9c47ac6b-ef64-497b-aa5c-24544036f42b)
![image](https://github.com/zulip/zulip/assets/2343554/16fb53b8-6c63-4b34-8efb-80e35372bfbf)

- https://zulip.com/help/search-for-messages#search-for-attachments-or-links
![image](https://github.com/zulip/zulip/assets/2343554/ffca2519-d9bd-472e-b785-85805ee0f9f9)

- https://zulip.com/help/share-and-upload-files
![image](https://github.com/zulip/zulip/assets/2343554/c5ca7ef0-4223-4752-aef9-0728b2223dd3)

alya commented:
Thanks! I think we can make this look a bit more standard my having instructions to:

1. open the viewer
2. interact with the image, with the bulleted list being under this instruction
3. close the viewer when done

alya commented:
I think the intro needs one more sentence, something like:

"You can open the **image viewer** to see the image in more detail or download it."

alya commented:
And let's split out browsing other images in the view into it's own section and set of instructions. It's a pretty cool feature, and worth calling out separately.

drrosa commented:
@alya I was thinking maybe we should add in the intro that the viewer supports web images and YouTube videos? Something like:

> When someone pastes or attaches an image to a message or adds a link to a web image or YouTube video, Zulip shows a small preview. You can open the image viewer to see an image in more detail, download it, or watch a YouTube video.

alya commented on help/view-and-browse-images.md:
```diff
@@ -1,36 +1,83 @@
+{tab|desktop-web}
 
-* **Reset zoom** so that the image is recentered and its original size
+1. Click on the image preview to open the **image viewer**.
```
Maybe "an image preview"?

alya commented on help/view-and-browse-images.md:
```diff
@@ -1,36 +1,83 @@
+1. Click on the image preview to open the **image viewer**.
 
-* **Open** the image in a new browser tab
+1. Interact with the image.
```
Maybe add: "You can:"? We aren't really telling users to do *all* of these things every time.

alya replied:
(same goes for mobile)

alya commented on help/view-and-browse-images.md:
```diff
@@ -1,36 +1,83 @@
 
-To open the image viewer, you must click on the image preview, not the file name.
+To open the image viewer in the Zulip desktop or web app, you must click on the
+image preview, not the file name.
```
How about a positive framing on this? Something like: 

> You can click on the image file name to download it rather than viewing it in the Zulip app.

We could make this a tip on the the web/desktop instructions for viewing an image, and drop this section.

alya commented:
> @alya I was thinking maybe we should add in the intro that the viewer supports web images and YouTube videos? Something like:

Mm, yeah, we should describe interactions with videos, either here or on a dedicated page. I believe #20449 is expected to be completed pretty soon, so it probably makes sense to wait on that before we make adjustments.

alya commented on help/view-and-browse-images.md:
```diff
@@ -1,36 +1,83 @@
+## Using the image browser
+
+In the Zulip desktop or web app, you can browse the images in the current view.
+For example, if you're in a stream view, the image browser will show all the
```
Let's use a conversation view as the example here, rather than a stream view. We're generally moving in the direction of reducing the prominence of interleaved stream views.

alya commented:
https://github.com/zulip/zulip/pull/20449 has been merged, so I think we can update this page to talk about images and videos. Maybe rename it to "View images and videos"?

drrosa commented:
Thanks for the feedback! I updated the PR and screenshots. I updated the page title and intro to include videos, but I haven't described interactions with videos in the other sections yet.

drrosa commented:
@zulipbot add "help center review"

drrosa commented:
@alya I updated the PR + screenshots.

alya commented:
Hm, after https://github.com/zulip/zulip-mobile/issues/5781, videos will open in an external browser on mobile. We probably don't want to document any details about the possible interactions, since they'll be system-dependent?

drrosa commented:
> Hm, after [zulip/zulip-mobile#5781](https://github.com/zulip/zulip-mobile/issues/5781), videos will open in an external browser on mobile. We probably don't want to document any details about the possible interactions, since they'll be system-dependent?

I think it would still be nice to advertise that the video interactions on desktop/web will also possible on mobile since we can expect everything to work the same on major browsers. But maybe it makes more sense to change the mobile documentation to something like:

> 1. Tap a video thumbnail, link, or file name to open the video in your device's default browser.
> 
> 1. You can switch back to the Zulip app once you are done.

alya commented:
Yeah, that seems fine. I just don't think we should be in the business of documenting external browsers' capabilities.

drrosa commented:
Updated PR + screenshot.

alya commented on help/view-and-browse-images.md:
```diff
@@ -1,36 +1,67 @@
+small preview. Similarly, links to images on the web or YouTube videos will
+also generate a small preview. You can open the image viewer to see an image
+in more detail, download it, or watch a video.
 
```
I think the intro needs to be adjusted in light of https://zulip.com/help/allow-image-link-previews (previews can be turned off).

Could you please check whether video previews are controlled by the same setting? https://zulip.com/help/allow-image-link-previews should probably be updated as well.

alya replied:
Oh, and we actually need to tweak the instructions for opening the preview to something like this, since there might not be a preview:

> Click an image name or preview to open the **image viewer**.

drrosa replied:
> Could you please check whether video previews are controlled by the same setting? https://zulip.com/help/allow-image-link-previews should probably be updated as well.

I checked and it turns out that only YouTube videos are controlled by the same setting. Video previous are always shown for uploaded videos regardless of whether the setting is enabled or disabled.

drrosa replied:
> Oh, and we actually need to tweak the instructions for opening the preview to something like this, since there might not be a preview:
> 
> > Click an image name or preview to open the **image viewer**.

I double checked, and it looks like the user has to click on an image preview to open the image viewer. Otherwise, the image will open up in a new tab.

alya replied:
Huh, I'm confused now why I thought that worked.

So maybe there's no way to get to the image viewer if previews are turned off? I tried `V`, but that didn't seem to work either.

alya replied:
> I checked and it turns out that only YouTube videos are controlled by the same setting. Video previews are always shown for uploaded videos regardless of whether the setting is enabled or disabled.

Oh, this sounds like a bug. I'll flag it in #issues and mention you. But we should just document it as if the setting controls all image and video previews, as intended.

> Similarly, links to images on the web or YouTube videos will also generate a small preview [if this setting is turned on](https://zulip.com/help/allow-image-link-previews)...

I have a vague recollection that language we use elsewhere is more like:

> Similarly, links to images and videos will also generate a small preview, unless previews are [disabled](https://zulip.com/help/allow-image-link-previews) in your organization.

alya replied:
Ok, I also confirmed with @timabbott that there is intentionally no lightbox if image previews are turned off. So let's just make sure the documentation is written to be consistent with that, without being too distracting for the more common situation where image previews are enabled.

drrosa commented:
@alya I updated the PR + screenshots to address your feedback. Also, I updated the filename to /help/view-images-and-videos page and tweaked the documentation on pages linking to it.

alya commented:
Thanks! This article should be cross-linked with /help/allow-image-link-previews.

alya commented on help/view-images-and-videos.md:
```diff
@@ -0,0 +1,137 @@
+
+1. Click an image preview or video thumbnail to open the viewer.
+
+1. Use the left and right arrow buttons at the bottom of the viewer.
```
Maybe:

Browse by using the left and right arrow buttons at the bottom of the viewer.

alya commented on help/view-images-and-videos.md:
```diff
@@ -0,0 +1,137 @@
+1. Use the left and right arrow buttons at the bottom of the viewer.
+
+1. Click any image or video to display it in viewer.
+
```
How about just this?

> Click any image or video to display it.

alya commented on help/view-images-and-videos.md:
```diff
@@ -0,0 +1,137 @@
+a small preview. Similarly, links to images and videos will also generate a
+small preview, unless previews are [disabled](/help/allow-image-link-previews)
+in your organization. You can open the image viewer to see an image in more
+detail, download it, or watch a video.
```
How about something like this instead?

Zulip shows previews of attached images and videos, unless previews are [disabled](/help/allow-image-link-previews) in your organization. You can click on a preview to view an image in more detail, or to play a video. Zulip also makes it convenient to browse all images and videos attached to messages in your current view.

drrosa commented:
@alya I updated the PR + screenshots to address your feedback.

alya commented:
Works for me!

timabbott commented:
Merged, thanks @drrosa!
